CrowServe/CrowServe.h

```diff
@@ -32,8 +55,18 @@ struct MessageHeader {
 
 #pragma pack(pop)
 
+bool initialize_sockets();
+void shutdown_sockets();
+
 class Socket {
 public:
+    Socket() {
+        initialize_sockets();
+    }
+
+    ~Socket() {
+        shutdown_sockets();
```

Comment:
These are wrong, since you can have multiple sockets.

Also prefer to not define destructor if you don't have to, since then you sort off have to follow the rule of 5, unless you want to have bad hard to spot bugs. (We probably should delete copy and move anyways..)

It's probably better to just expect the lib user to call these, or we could have a global ref counter (although that might fuck with the user, if they init the libs themselves, so probably bad idea).

Missing separation line after destructor.

Reply:
Winsock keeps track of calls to WSAStartup and WSACleanup and you can call each multiple times as long as they're balanced (i.e. one WSACleanup for each WSAStartup)

Reply:
I removed constructor & destructor and call init/deinit in the test case instead for now